follow_fork/follow_fork.c

```diff
@@ -0,0 +1,124 @@
+/**
+ * Allows the LiveRecorder API to follow fork() and record
+ * all children processes.
+ *
+ * See README.md for how to compile and use.
+ *
+ */
+#define _POSIX_SOURCE
+#define _GNU_SOURCE
+
+#include "undolr.h"
+#include "undolr_deprecated.h"
+
+#include <errno.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdbool.h>
+#include <time.h>
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <linux/limits.h>
+
+extern char *__progname;
+
+void attach_lr(int, char**, char**) __attribute__ ((constructor));
+
+static bool
+process_needs_recording(pid_t pid)
+{
+    if (strncmp(__progname, "undo", 4) == 0)
+    {
+        if (unsetenv("LD_PRELOAD") == -1)
+        {
+            perror("unsetenv");
+            exit(1);
+        }
+        return false;
+    }
+    return true;
+}
+
+static void *
+get_sym_addr(const char *sym)
+{
+    void *fptr = NULL;
+
+    fptr = dlsym(RTLD_NEXT, sym);
+    if (fptr == NULL)
+    {
+        fprintf(stderr, "Error: %s, calling dlsym with %s\n",
+                        dlerror(), sym);
+        abort();
+    }
+    return fptr;
+}
+
+static void
+start_recording(pid_t pid)
```

Comment:
I think that the pid should not be passed in as it's error prone and doesn't save any significat amount of time. Just do `getpid()` in this function! :)

Reply:
OK, sounds good